src/styles/_catalog.scss

```diff
@@ -1,22 +1,37 @@
 .catalog {
-  display: flex;
-  flex-wrap: wrap;
+  display: grid;
+  justify-content: center;
+  grid-template-columns: repeat(4, 200px);
+  gap: 48px;
 
   max-width: 944px;
   margin: 0 auto;
 
   &__card {
-    display: flex;
+    display: grid;
     justify-content: center;
     align-items: center;
     width: 200px;
     height: 300px;
     border: 1px solid #ccc;
     border-radius: 5px;
-    margin-bottom: 48px;
   }
+}
+
+@media screen and (max-width: 1023px) {
+  .catalog {
+    grid-template-columns: repeat(3, 200px);
+  }
+}
+
+@media screen and (max-width: 767px) {
+  .catalog {
+    grid-template-columns: repeat(2, 200px);
+  }
+}
 
-  &__card:not(:nth-child(4n)) {
-    margin-right: 48px;
+@media screen and (max-width: 487px) {
+  .catalog {
+    grid-template-columns: repeat(1, 200px);
```

Comment:
for sake of practice you could try to generate all three @media using @each loop.
However it's not required.

Reply:
Hi Amit, I have created all 3 @media queries via @each loop. Let me know if it looks ok. Thanks!